include/seqan3/search/algorithm/detail/search_trivial.hpp

```diff
@@ -56,7 +56,7 @@ namespace seqan3::detail
 
 /*!\brief Searches a query sequence in an index using trivial backtracking.
  * \tparam abort_on_hit  If the flag is set, the search algorithm aborts on the first hit.
- * \tparam iterator_t    Must model seqan3::FmIndexIterator.
+ * \tparam iterator_t    Must model seqan3::FmIndexCursor.
  * \tparam query_t       Must be a std::ranges::InputRange over the index's alphabet.
  * \tparam delegate_t    Takes `index::iterator_type` as argument.
  * \param[in] it         Iterator of atring index built on the text that will be searched.
```

Comment:
I am not sure if `it` needs to be renamed to `cur` or something like that, but I guess it can't hurt?